src/java/org/apache/cassandra/index/sai/IndexContext.java

```diff
@@ -164,6 +172,7 @@ public IndexContext(String keyspace,
                                     ? AbstractAnalyzer.fromOptionsQueryAnalyzer(getValidator(), options)
                                     : this.analyzerFactory;
         this.primaryKeyFactory = Version.LATEST.onDiskFormat().primaryKeyFactory(clusteringComparator);
+        this.segmentCompactionEnabled = true;
```

Comment:
For this constructor, that is only used in testing, it seems that we could pass whether compaction is enabled in in the constructor

Reply:
I tried to do that, but it turned out to be a lot more work than I thought, because `QueryController#getContext` is used also in `Operation#analyzeGroup`. Let's not hold this ticket.